net-utils/src/token_bucket.rs

```diff
@@ -88,6 +88,24 @@ impl TokenBucket {
         }
     }
 
+    /// Returns time in microseconds until `num_tokens` worth of new
+    /// tokens can be minted.
+    ///
+    /// Calculation is performed assuming no demand for smaller
+    /// batches of tokens (actual time may be longer).
+    /// Returns None if num_tokens > bucket capacity.
+    #[inline]
+    pub fn us_to_have_tokens(&self, num_tokens: u64) -> Option<u64> {
```

Comment:
What is the num_tokens? Additional tokens? Or the target total tokens?

Reply:
The comment above was not ideal (fixed), but it is the number of tokens to be present in the bucket.

Reply:
if it is the number of tokens to be present in the bucket, I am puzzled by the math, should not it be 
num_tokens - current_tokens instead?

if current_tokens >= num_tokens return 0 as desired tokens already available, otherwise

wait for num_tokens - current_tokens/(token refresh rate)

Reply:
OMG yes it was backwards! Note to self: no new "trivial" code without unit tests.